sql/src/main/java/org/apache/druid/sql/calcite/expression/builtin/DateTruncOperatorConversion.java

```diff
@@ -93,14 +93,18 @@ public DruidExpression toDruidExpression(
           final Expr truncTypeExpr = plannerContext.parseExpression(inputExpressions.get(0).getExpression());
 
           if (!truncTypeExpr.isLiteral()) {
-            throw new IAE("Operator[%s] truncType must be a literal", calciteOperator().getName());
+            throw InvalidSqlInput.exception("Operator[%s] truncType must be a literal", calciteOperator().getName());
```

Comment:
Even better would be to add `.literalOperands(0)` to the operator conversion. Then it'll be a proper validation error, and we won't need this check here, because it wouldn't be able to happen.

Reply:
Makes sense. Done.